socorro/processor/cache_manager.py

```diff
@@ -259,7 +270,10 @@ def _event_generator(self, nonblocking=False):
         try:
             while self.running:
                 try:
-                    for event in self.inotify.read(timeout=timeout):
+                    events = self.inotify.read(timeout=timeout)
```

Comment:
The `inotify.read` docs say it returns a generator, but that's a lie--it's returning a list. If they change it to return a generator, calling `.pop` and `.insert` on it will fail and we'll know immediately in CI.